pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -149,8 +150,8 @@ type OpenStack struct {
 	localInstanceID string
 }
 
-// Config is used to read and store information from the cloud configuration file
-type Config struct {
+// Workaround is used to read and store information from the cloud configuration file
+type Workaround struct {
```

Comment:
I'd rather try to get rid of this struct entirely. Think of this workflow:

1) Attempt to load this configuration (warn the user that it'll be deprecated if a config like this is found)
2) If found, load the data directly into an `AuthOpts` object
3) If not found, attempt to load the configs using `clientconfig`

this way we preserve backwards compatibility, we reduce the code and we move the internal structures to using `clientconfig` and the upstream standard

Reply:
Yeah, the issue is that in order to read from the custom config file, we need a struct with the gcgf annotations, so we will have to have something similar to the workaround struct until we fully phase out config files. Other than that though, this makes sense.